### PR TITLE
Refactor dependency management for readability, clarity on theoretical example

### PR DESCRIPTION
**Pull request by nathan-contino**
- https://github.com/gradle/gradle/issues/20640
- Readability improvements across the board

### Context
- Rephrases "example project" explanation to avoid confusion (there is no project, just a graphic)
- General best-practice documentation sentence and section rewrites for clarity and readability

### Current Page
https://docs.gradle.org/current/userguide/core_dependency_management.html

### Rendered Docs Preview of Changes
https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/59052397:id/distributions/gradle-8.0-docs.zip!/gradle-8.0-20221202201707%2B0000/docs/userguide/core_dependency_management.html

### Contributor Checklist
- [X] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [X] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [X] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [X] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [X] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [X] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [X] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [X] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [X] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/core_dependency_management.adoc [hunk 1]
```diff
@@ -1,60 +1,60 @@
...
 
 [[sec:dependency-mgmt-in-gradle]]
-== Dependency management in Gradle
+== Dependency Management in Gradle
 
-Gradle has built-in support for dependency management and lives up to the task of fulfilling typical scenarios encountered in modern software projects.
-We’ll explore the main concepts with the help of an example project.
-The illustration below should give you an rough overview on all the moving parts.
+Gradle has built-in support for dependency management.
+We’ll explore the main concepts with the help of the theoretical example project illustrated below:
```
Comment (hythloda) [inline subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/core_dependency_management.adoc:18]: question: should we just remove the word example per @jbartok suggestion?
```suggestion
We’ll explore the main concepts with the help of the theoretical project illustrated below:
```

Comment (nathan-contino) [reply subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/core_dependency_management.adoc:18]: Good point, I think "theoretical project" does enough work here on its own. Good example of a word we can eliminate with a bonus of eliminating confusion.

**Comment by hythloda** [comment]
Copy edit - LGTM
Much improved readability

**Comment by nathan-contino** [comment]
@bot-gradle test BD

**Comment by asodja** [comment]
I find it confusing that there is `theoretical project illustrated below:` and below that sentence, there is an image `Dependency management big picture` that has no relation to that theoretical project. And the whole explanation with a `theoretical project` is confusing as well, too abstract. But it's much better after the update, the previous version was really hard to follow and included whole dependencies terminology.

**Comment by nathan-contino** [comment]
@asodja good point, the image here really doesn't describe much. Do you think it adds value, or should we remove it from the page?

**Comment by asodja** [comment]
I think it adds some value, just the position is incorrect so it makes the example confusing. 

But there could be a short paragraph, where elements of the picture are explained (I guess it shows how Gradle fetches dependencies from repositories and where it stores them locally).

**Review by asodja — APPROVED** [review]
LGTM

**Comment by hythloda** [comment]
@bot-gradle test and merge
